src/csharp/Pester/CodeCoverageConfiguration.cs

```diff
@@ -57,17 +57,16 @@ public CodeCoverageConfiguration(IDictionary configuration) : this()
         {
             if (configuration != null)
             {
-                Enabled = configuration.GetValueOrNull<bool>("Enabled") ?? Enabled;
-                OutputFormat = configuration.GetObjectOrNull<string>("OutputFormat") ?? OutputFormat;
-                OutputPath = configuration.GetObjectOrNull<string>("OutputPath") ?? OutputPath;
-                OutputEncoding = configuration.GetObjectOrNull<string>("OutputEncoding") ?? OutputEncoding;
-                Path = configuration.GetArrayOrNull<string>("Path") ?? Path;
-                ExcludeTests = configuration.GetValueOrNull<bool>("ExcludeTests") ?? ExcludeTests;
-                RecursePaths = configuration.GetValueOrNull<bool>("RecursePaths") ?? RecursePaths;
-                CoveragePercentTarget = configuration.GetValueOrNull<decimal>("CoveragePercentTarget") ?? CoveragePercentTarget;
-
-                SingleHitBreakpoints = configuration.GetValueOrNull<bool>("SingleHitBreakpoints") ?? SingleHitBreakpoints;
-                UseBreakpoints = configuration.GetValueOrNull<bool>("UseBreakpoints") ?? UseBreakpoints;
+                if (configuration.GetValueOrNull<bool>(nameof(Enabled)) is bool enabled) { Enabled = enabled; };
```

Comment:
This looks like a lot of repeated code that can be done wrong in one or two places. I would rather see this in a method, and using `is not null` or `!= null` rather than `is <type>`.

IMHO you are checking to see if the value is not null, because GetValueOrNull already guarantees the given T type is returned.

Nice to see return to nameof, I had that everywhere but for some reason that I don't remember it did not work, and I reverted to just names. Maybe I was trying to avoid shipping dll, and compile in-line.

```cs
configuration.AssignValueIfNotNull<bool>(nameof(Enabled), v => Enabled = v);

public static void AssignValueIfNotNull<T>(this IDictionary dictionary, string key, Action<T> assign) 
where T : struct
{
    var value = GetValueOrNull<T>(dictionary, key);
    if (value != null)
    {
        assign(value.Value);
    }
}

```

Reply:
> IMHO you are checking to see if the value is not null, because GetValueOrNull already guarantees the given T type is returned.

Fair point. Shortsighted in the heat of the moment. Had to get something to work for the Ansi-option which I'll now probably skip. Will still fix this though for future use.

Love the generic extension-method. 💯 will steal 😄